x/contractmanager/keeper/msg_server.go

```diff
@@ -40,3 +40,32 @@ func (k Keeper) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams)
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+// ResubmitFailure resubmits the failure after contract acknowledgement failed
+func (k Keeper) ResubmitFailure(goCtx context.Context, req *types.MsgResubmitFailure) (*types.MsgResubmitFailureResponse, error) {
```

Comment:
why do we have `ResubmitFailure` and `DoResubmitFailure`? what's the difference between them? can we unify so there's only one Keeper's method for failures resubmission?